core/app/models/spree/refund.rb

```diff
@@ -10,13 +10,14 @@ class Refund < Spree::Base
 
     validates :payment, presence: true
     validates :reason, presence: true
-    validates :transaction_id, presence: true, on: :update # can't require this on create because the before_create needs to run first
     validates :amount, presence: true, numericality: { greater_than: 0 }
 
     validate :amount_is_less_than_or_equal_to_allowed_amount, on: :create
 
+    attr_accessor :perform_after_creation
+    after_create :set_perform_after_creation_default
     after_create :perform!
-    after_create :create_log_entry
```

Comment:
Was this moved somewhere else or simply removed? I couldn't find it in the PR so I'm assuming the latter?

Reply:
This is the code of that method:

```
    def create_log_entry
      log_entries.create!(details: @response.to_yaml)
    end
```

but `@response` is only populated during [`perform!` execution](https://github.com/solidusio/solidus/blob/ac325ef25194a61bf97a7ad975cd50853ac921ce/core/app/models/spree/refund.rb#L49). This means that if `perform!` is not called that LogEntry was empty. That stuff has been moved into the `perform!` method and `create_log_entry` is a private method so I thought it was ok just getting rid of that. Do you think we need to keep and deprecate it instead?

Reply:
I think it's good, I didn't realize you had moved it to `#perform!`.